operator/pkg/appstate/appstate.go

```diff
@@ -187,18 +184,37 @@ func (m *AppMonitor) runInformers(ctx context.Context, informers []types.StatusI
 		close(resourceStateCh)
 	}()
 
-	goRun := func(fn runControllerFunc) {
+	// Collect namespace/kind pairs
+	namespaceKinds := make(map[string]map[string][]types.StatusInformer)
+	for _, informer := range informers {
+		kindsInNs, ok := namespaceKinds[informer.Namespace]
+		if !ok {
+			kindsInNs = make(map[string][]types.StatusInformer)
+		}
+		kindsInNs[informer.Kind] = append(kindsInNs[informer.Kind], informer)
+		namespaceKinds[informer.Namespace] = kindsInNs
+	}
+
+	goRun := func(fn runControllerFunc, namespace string, informers []types.StatusInformer) {
 		shutdown.Add(1)
 		go func() {
-			fn(ctx, m.clientset, m.targetNamespace, informers, resourceStateCh)
+			fn(ctx, m.clientset, namespace, informers, resourceStateCh)
 			shutdown.Done()
 		}()
 	}
-	goRun(runDeploymentController)
-	goRun(runIngressController)
-	goRun(runPersistentVolumeClaimController)
-	goRun(runServiceController)
-	goRun(runStatefulSetController)
+
+	kindImpls := map[string]runControllerFunc{
+		DeploymentResourceKind:            runDeploymentController,
+		IngressResourceKind:               runIngressController,
+		PersistentVolumeClaimResourceKind: runPersistentVolumeClaimController,
+		ServiceResourceKind:               runServiceController,
+		StatefulSetResourceKind:           runStatefulSetController,
+	}
+	for namespace, kinds := range namespaceKinds {
+		for kind, informers := range kinds {
+			goRun(kindImpls[kind], namespace, informers)
```

Comment:
This needs a check that `kindImpls[kind]` exists, and log something with `kind` string in it if it does not.  Otherwise, this is great.

Reply:
Do you want an issue to make sure this gets into release notes?

Reply:
Added.